sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,602 @@
+"""
+    y_cable_base.py
+
+    Base class for implementing Y-Cable specific functionality in SONiC.
+    This is the base class of sonic_y_cable implentation. A vendor specific 
+    implementation of YCable needs to inherit from this class.
+    Classes derived from this class provide the ability to interact
+    with a vendor specific Y-Cable
+"""
+
+
+class YCableBase(object):
+
+    # definitions of targets for getting the various fields/cursor
+    # equalization parameters from the register spec
+    # the name of the target denotes which side MCU
+    # values will be retreived on the Y-Cable
+
+    TARGET_UNKNOWN = -1
+    TARGET_NIC = 0
+    TARGET_TOR_A = 1
+    TARGET_TOR_B = 2
+
+    # definitions of targets for getting the EYE/BER
+    # and initiating PRBS/Loopback on the Y cable
+    # the name of the target denotes which side values
+    # will be retreived/initiated
+
+    EYE_PRBS_TARGET_LOCAL = 0
+    EYE_PRBS_TARGET_TOR_A = 1
+    EYE_PRBS_TARGET_TOR_B = 2
+    EYE_PRBS_TARGET_NIC = 3
+
+    # definitions of switch counter types
+    # to be entered by the user in get_switch_count api
+    # for retreiving the counter values
+
+    SWITCH_COUNT_MANUAL = "manual"
+    SWITCH_COUNT_AUTO = "auto"
+
+    # switching modes inside muxcable
+    SWITCHING_MODE_MANUAL = 0
+    SWITCHING_MODE_AUTO = 1
+
+    # Valid return codes for upgrade firmware routine steps
+    FIRMWARE_DOWNLOAD_SUCCESS = 0
+    FIRMWARE_DOWNLOAD_FAILURE = 1
+    FIRMWARE_ACTIVATE_SUCCESS = 0
+    FIRMWARE_ACTIVATE_FAILURE = 1
+    FIRMWARE_ROLLBACK_SUCCESS = 0
+    FIRMWARE_ROLLBACK_FAILURE = 1
+
+    def __init__(self, port):
+        self.port = port
+
+    def toggle_mux_to_torA(self, port):
```

Comment:
Do we need to pass `port` to all these functions? The port is passed in in the initializer; therefore the vendor can simply refer to `self.port`.

Reply:
fixed, removed from all the API's